src/qbe_ir_generator.cpp

```diff
@@ -124,14 +124,28 @@ void QbeIrGenerator::Visit(const IfStmtNode& if_stmt) {
   if_stmt.predicate->Accept(*this);
   int predicate_num = num_recorder.NumOfPrevExpr();
   int label_num = NextLabelNum();
-  std::string start_label = PrefixLabel("if", label_num);
+  std::string if_label = PrefixLabel("if", label_num);
+  std::string else_label = PrefixLabel("else", label_num);
   std::string end_label = PrefixLabel("end", label_num);
 
-  // Jumps to start_label if argument is non-zero. Or else, jumps to end_label.
-  output << "jnz " << PrefixSigil(predicate_num) << ", " << start_label << ", "
-         << end_label << std::endl;
-  output << start_label << std::endl;
-  if_stmt.body->Accept(*this);
+  // Jumps to if_label if argument is non-zero. Or else, jumps to the second
+  // label.
```

Comment:
I would prefer a comment explaining the logic of the generated code instead of attempting to rephrase each line individually:
```suggestion
  // Jumps to "then" if the predicate is true (non-zero), else jumps to "else".
  // If no "else" part exists, jumps directly to "end".
  // In the presence of an "else" part, a second jump is needed after executing
  // "then" to skip it, as the generated code for "else" follows immediately
  // after "then".
```

Reply:
Great suggestions! 👍🏼